From eaf3ca398df29816d4453876e0575ca344f91e73 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 23 Sep 2025 03:04:44 +0200
Subject: [PATCH] Don't use Typst engine

The Typst engine treats any URLs as relative paths, breaking all external
resources. Use the XeTeX engine instead for now.
---
 apostrophe/export_dialog.py | 5 ++---
 apostrophe/main_window.py   | 2 --
 data/media/formats.json     | 6 ------
 3 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/apostrophe/export_dialog.py b/apostrophe/export_dialog.py
index f3ab16b..ea5a5bd 100644
--- a/apostrophe/export_dialog.py
+++ b/apostrophe/export_dialog.py
@@ -83,9 +83,8 @@ class ExportDialog:
             "extension": "pdf",
             "to": "pdf",
             "mimetype": "application/pdf",
-            "args": ["--pdf-engine=typst",
-                     "--variable=papersize:a4",
-                     "-V", "template=%s" % helpers.get_media_path('/pandoc_templates/typst.typ')]
+            "args": ["--pdf-engine=xelatex",
+                     "--variable=papersize:a4"]
         },
         "html":
         {
diff --git a/apostrophe/main_window.py b/apostrophe/main_window.py
index 7d73aac..6d43bcf 100644
--- a/apostrophe/main_window.py
+++ b/apostrophe/main_window.py
@@ -88,8 +88,6 @@ class MainWindow(Adw.ApplicationWindow):
         super().__init__(application=Gio.Application.get_default(),
                          title="Apostrophe")
 
-        os.chdir("/")
-
         # Preferences
         self.settings = Settings.new()
 
diff --git a/data/media/formats.json b/data/media/formats.json
index 07cb846..6de92a6 100644
--- a/data/media/formats.json
+++ b/data/media/formats.json
@@ -3,12 +3,6 @@
       "name": "PDF",
       "ext": "pdf",
       "to": "pdf",
-      "engine": "typst"
-    },
-    {
-      "name": "PDF (LaTeX)",
-      "ext": "pdf",
-      "to": "pdf",
       "engine": "xelatex"
     },
     {
-- 
2.51.0

